it seems on some of our targets, the c and c++ vulkan types are not implicitly convertible

--- a/gtk/src/gtk_display_driver_vulkan.cpp
+++ b/gtk/src/gtk_display_driver_vulkan.cpp
@@ -66,12 +66,12 @@ bool S9xVulkanDisplayDriver::init_imgui()
     init_info.Device = context->device;;
     init_info.QueueFamily = context->graphics_queue_family_index;
     init_info.Queue = context->queue;
-    init_info.DescriptorPool = imgui_descriptor_pool.get();
+    init_info.DescriptorPool = (VkDescriptorPool)imgui_descriptor_pool.get();
     init_info.Subpass = 0;
     init_info.MinImageCount = context->swapchain->get_num_frames();
     init_info.ImageCount = context->swapchain->get_num_frames();
     init_info.MSAASamples = VK_SAMPLE_COUNT_1_BIT;
-    ImGui_ImplVulkan_Init(&init_info, context->swapchain->get_render_pass());
+    ImGui_ImplVulkan_Init(&init_info, (VkRenderPass)context->swapchain->get_render_pass());
 
     auto cmd = context->begin_cmd_buffer();
     ImGui_ImplVulkan_CreateFontsTexture(cmd);
